src/pid.cpp

```diff
@@ -179,16 +179,20 @@ double Pid::compute_command(
 
 double Pid::compute_command(double error, double error_dot, const double & dt_s)
 {
+  // don't update anything
+  if (dt_s <= 0.0) {
+    return cmd_;
+  }
+
   // Get the gain parameters from the realtime buffer
   Gains gains = *gains_buffer_.readFromRT();
 
   double p_term, d_term;
   p_error_ = error;  // this is error = target - state
   d_error_ = error_dot;
 
-  if (
-    dt_s <= 0.0 || !std::isfinite(error) || !std::isfinite(error_dot)) {
-    return 0.0;
+  if (!std::isfinite(error) || !std::isfinite(error_dot)) {
+    return cmd_ = i_term_ = 0.0;
```

Comment:
Do you need to do this?

Because iterm is only set later

Reply:
If we reset the cmd here, I agree with
https://github.com/ros-controls/control_toolbox/issues/31#issuecomment-59842858

we can also set both to NaN and just proceed. But the PID can't recover anyways then

Reply:
I'm not sure at this point. If it's not finite, i personally prefer the output to be NaN, instead of Zero. Zero is still a valid value

Reply:
What do you think of this?

Reply:
so you propose return NaN and reset the internal states, and it recovers the next time having a valid input?

Reply:
I saw only return NaN and not reset internal states, as this can happen when one of the reading is NaN and we don't want the i_term to reset and it takes a time for the system to stabilize